### ISSUE TITLE
Opening and saving .mp3 corrupts file

### ISSUE THREAD

**Issue by Vesper-Works** [mono/taglib-sharp#237]
Installing from NuGet, both TagLibSharp and TagLibSharp Netstandard2.0 corrupts the .mp3 file after simply opening the file and saving it. 
Hex data of the file:
Before:
![image](https://user-images.githubusercontent.com/59376295/98857089-7560d600-2456-11eb-9fda-c139fb6a4d04.png)

After:
![image](https://user-images.githubusercontent.com/59376295/98857144-90334a80-2456-11eb-81af-351c90a351a2.png)
![image](https://user-images.githubusercontent.com/59376295/98857171-988b8580-2456-11eb-8f0d-06635c266a3f.png)

Also, the metadata of the file isn't changed either:
![image](https://user-images.githubusercontent.com/59376295/98857351-dab4c700-2456-11eb-8b99-591b851ac105.png)

```
                    await youtube.Videos.Streams.DownloadAsync(streamInfo, Path.Combine(path, author + " - " + title + ".mp3"));

                    TagLib.File f = TagLib.File.Create(Path.Combine(path, author + " - " + title + ".mp3"));
                    f.Tag.Performers = new string[] { author };
                    f.Tag.Title = title;
                    f.Save();
```

Is this a problem anyone else has faced or am I misunderstanding something? 
(regardless of what goes into the metadata, the result is still the same).

**Comment by tgreve15** [comment]
Doesn't the first line of your code suggest it's downloading the file
asynchronously? Are you sure that's finished before Taglib kicks in?
Or does the await command ensure that is finished?
Have you tried it with the file local and run the taglib bit?

**Comment by Vesper-Works** [comment]
@tgreve15 Testing the code on an mp3 file I already have reveals that the code is working, the tags change as I set them perfectly.
Running the code on button press, long after I've finished downloading the .mp3 to make sure, still corrupts the file.

I reckon the issue is with the file download not being mp3, which is screwing things up.
I'll add an ffmpeg conversion to mp3 after finding out what the file natively comes as to solve it, thank you for your helpful questions, I would be and have been stuck on this for a while!
